From e002f976000ecdea1d8e3893520aaef12d54a56e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Thu, 21 Feb 2019 08:55:22 +0100
Subject: [PATCH 1/1] Move environment helpers to module
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This way they can be reused in the docker settings as well.

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 weblate/settings_openshift.py           |  7 +---
 weblate/utils/environment.py            | 41 +++++++++++++++++++++
 weblate/utils/tests/test_environment.py | 48 +++++++++++++++++++++++++
 3 files changed, 90 insertions(+), 6 deletions(-)
 create mode 100644 weblate/utils/environment.py
 create mode 100644 weblate/utils/tests/test_environment.py

diff --git a/weblate/settings_openshift.py b/weblate/settings_openshift.py
index ce95ce0140..102d19d602 100644
--- a/weblate/settings_openshift.py
+++ b/weblate/settings_openshift.py
@@ -25,12 +25,7 @@ from weblate.openshiftlib import get_openshift_secret_key, import_env_vars
 # Import example settings file to get default values for Weblate settings.
 from weblate.settings_example import *  # noqa
 
-
-def get_env_list(name, default=None):
-    """Helper to get list from environment."""
-    if name not in os.environ:
-        return default or []
-    return os.environ[name].split(',')
+from weblate.utils.environment import get_env_list
 
 
 DEBUG = False
diff --git a/weblate/utils/environment.py b/weblate/utils/environment.py
new file mode 100644
index 0000000000..03e6b7d20c
--- /dev/null
+++ b/weblate/utils/environment.py
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2012 - 2019 Michal Čihař <michal@cihar.com>
+#
+# This file is part of Weblate <https://weblate.org/>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from __future__ import unicode_literals
+import os
+
+
+def get_env_list(name, default=None):
+    """Helper to get list from environment."""
+    if name not in os.environ:
+        return default or []
+    return os.environ[name].split(',')
+
+
+def get_env_map(name, default=None):
+    """
+    Helper to get mapping from environment.
+
+    parses 'full_name:name,email:mail'
+    into {'email': 'mail', 'full_name': 'name'}
+    """
+    if os.environ.get(name):
+        return dict(e.split(':') for e in os.environ[name].split(','))
+    return default or {}
diff --git a/weblate/utils/tests/test_environment.py b/weblate/utils/tests/test_environment.py
new file mode 100644
index 0000000000..a209426316
--- /dev/null
+++ b/weblate/utils/tests/test_environment.py
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2012 - 2019 Michal Čihař <michal@cihar.com>
+#
+# This file is part of Weblate <https://weblate.org/>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+from __future__ import unicode_literals
+
+import os
+
+from django.test import SimpleTestCase
+
+from weblate.utils.environment import get_env_list, get_env_map
+
+class EnvTest(SimpleTestCase):
+    def test_list(self):
+        os.environ['TEST_DATA'] = 'foo,bar,baz'
+        self.assertEqual(get_env_list('TEST_DATA'), ['foo', 'bar', 'baz'])
+        os.environ['TEST_DATA'] = 'foo'
+        self.assertEqual(get_env_list('TEST_DATA'), ['foo'])
+        del os.environ['TEST_DATA']
+        self.assertEqual(get_env_list('TEST_DATA'), [])
+        self.assertEqual(get_env_list('TEST_DATA', ['x']), ['x'])
+
+    def test_map(self):
+        os.environ['TEST_DATA'] = 'foo:bar,baz:bag'
+        self.assertEqual(
+            get_env_map('TEST_DATA'), {'foo': 'bar', 'baz': 'bag'}
+        )
+        os.environ['TEST_DATA'] = 'foo:bar'
+        self.assertEqual(get_env_map('TEST_DATA'), {'foo': 'bar'})
+        del os.environ['TEST_DATA']
+        self.assertEqual(get_env_map('TEST_DATA'), {})
+        self.assertEqual(get_env_map('TEST_DATA', {'x': 'y'}), {'x': 'y'})
-- 
2.20.1

